Patches/StartOfRound.cs

```diff
@@ -5,58 +5,111 @@
+        {
+            if (!alreadyReplaced)
+            {
+                if (!foundInitial && instruction.opcode == OpCodes.Call && instruction.operand != null && instruction.operand.ToString() == "System.Collections.IEnumerator setPlayerToSpawnPosition(UnityEngine.Transform, UnityEngine.Vector3)")
+                {
+                    foundInitial = true;
+                }
+                else if (foundInitial && instruction.opcode == OpCodes.Ldc_I4_0)
+                {
+                    shouldSkip = true;
+                    continue;
+                }
+                else if (shouldSkip && instruction.opcode == OpCodes.Ldloc_0)
+                {
+                    shouldSkip = false;
+                    alreadyReplaced = true;
+                }
+            }
+
+            if (!shouldSkip)
+            {
+                newInstructions.Add(instruction);
+            }
+        }
+
+        if (!alreadyReplaced) Debug.LogError("Failed to transpile StartOfRound::OnPlayerConnectedClientRpc");
+
+        return newInstructions.AsEnumerable();
+    }
+
+    public static MethodInfo BeginSendClientRpc = typeof(RoundManager).GetMethod("__beginSendClientRpc", BindingFlags.NonPublic | BindingFlags.Instance);
+    public static MethodInfo EndSendClientRpc = typeof(RoundManager).GetMethod("__endSendClientRpc", BindingFlags.NonPublic | BindingFlags.Instance);
+
+    // Best guess at getting new players to load into the map after the game starts.
+    [HarmonyPostfix]
+    private static void Postfix(StartOfRound __instance, ulong clientId, int connectedPlayers, ulong[] connectedPlayerIdsOrdered, int assignedPlayerObjectId, int serverMoneyAmount, int levelID, int profitQuota, int timeUntilDeadline, int quotaFulfilled, int randomSeed) {
+	    if (__instance.connectedPlayersAmount + 1 >= __instance.allPlayerScripts.Length)
+		    Plugin.SetLobbyJoinable(false);
+
+        PlayerControllerB ply = __instance.allPlayerScripts[assignedPlayerObjectId];
+        // Make their player model visible.
+        ply.DisablePlayerModel(__instance.allPlayerObjects[assignedPlayerObjectId], true, true);
+
+        __instance.livingPlayers = __instance.connectedPlayersAmount + 1;
+        for (int i = 0; i < __instance.allPlayerScripts.Length; i++)
+        {
+            PlayerControllerB pcb = __instance.allPlayerScripts[i];
+            if (pcb.isPlayerControlled && pcb.isPlayerDead) __instance.livingPlayers--;
+        }
+
+        if (NetworkManager.Singleton != null)
+        {
+            for (int j = 0; j < __instance.connectedPlayersAmount + 1; j++)
+            {
+                // Don't set the player as controlled if they are dead.
+                if ((j == 0 || !__instance.allPlayerScripts[j].IsOwnedByServer) && !__instance.allPlayerScripts[j].isPlayerDead)
+                {
+                    __instance.allPlayerScripts[j].isPlayerControlled = true;
+                }
+            }
+        }
```

Comment:
```suggestion
        if (NetworkManager.Singleton != null)
        {
            for (int j = 0; j < __instance.connectedPlayersAmount + 1; j++)
            {
                // Don't set the player as controlled if they are dead.
                if ((j == 0 || !__instance.allPlayerScripts[j].IsOwnedByServer) && !__instance.allPlayerScripts[j].isPlayerDead)
                {
                    __instance.allPlayerScripts[j].isPlayerControlled = true;
                }
            }
        }
```

To make it easier for you to review this and the transpiler are the only additions.